go/common/crypto/signature/signer.go

```diff
@@ -201,7 +202,11 @@ func PrepareSignerContext(context Context) ([]byte, error) {
 		if chainContext == "" {
 			return nil, errNoChainContext
 		}
-		context = context + chainContextSeparator + chainContext
+		decodedChainContext, err := hex.DecodeString(string(chainContext))
```

Comment:
What are you doing here? There is no requirement that "chainContext" is a hex string. It can be an arbitrary string of 64 bytes.

For Ledger I believe we just restricted to a subset of ASCII, we could enforce that for contexts, but in this case such a check should be done at context registration (so it is caught immediately) not when things are being signed.

Reply:
Sorry I was just trying to reduce the length of the resulting context to test on the device. I'll revert this change now.